Framework/DataObjects/inc/MantidDataObjects/MDGridBox.tcc

```diff
@@ -1275,6 +1277,15 @@ TMDE(void MDGridBox)::integrateSphere(
       //        std::cout << "box at " << i << " has a vertex touching\n";
     }
 
+    // NOTE:
+    //  the smart partial box check (sub-devide box into smaller grid)
+    //  does not work for ellipsoid peaks, so we are force using the
+    //  finest grid for now until a suitable check can be identified
+    //  in the future.
+    if (isEllipsoidPeak) {
```

Comment:
Could we not do the partial box check but just using the major axis length for the spherical radius? Would that give a speedup?

Reply:
That's essentially what we are doing here. Given that this is a temporary solution (until we find a better way to check the partial box for ellipsoid) and checking partial box doesn't take a lot of time, this is the minimum amount of changes we can make to ensure it can work for both spherical peaks and ellipsoid peaks.